stable/yugabyte/templates/service.yaml

```
@@ -323,6 +323,12 @@ spec:
         {{- $_ := set $root.Values.tserver.affinity "podAntiAffinity" $podAntiAffinityData -}}
         {{ toYaml $root.Values.tserver.affinity | nindent 8 }}
         {{ end }}
+      {{- if $root.Values.dnsConfig }}
+      dnsConfig: {{- toYaml $root.Values.dnsConfig | nindent 8 }}
+      {{- end }}
```

We can use with here as well. Same for other occurrences. 
```suggestion
      {{- with $root.Values.dnsConfig }}
      dnsConfig: {{- toYaml . | nindent 8 }}
      {{- end }}
```

---

Updated with changes